Added Settings Menu

neoimpulse opened the pull request:
- Add a settings menu with the ability to turn the sound on and off.

neoimpulse committed:
Merge pull request #1 from neoimpulse/development

new pull request test

neoimpulse committed:
Revert "Revert "new pull request test""

This reverts commit cebcd76ce5647c7f6eeb0dde8e5a1d9d30d54f99.

neoimpulse committed:
Revert "new pull request test"

neoimpulse committed:
Merge pull request #2 from neoimpulse/revert-1-development

Revert "new pull request test"

dalarose commented on Assets/Scenes/mainGame.unity:
```diff
@@ -950,8 +950,7 @@ AudioSource:
   serializedVersion: 4
-  OutputAudioMixerGroup: {fileID: 24300002, guid: f2081dd5f9fdad449a9924db500054a5,
-    type: 2}
+  OutputAudioMixerGroup: {fileID: 0}
```
do you still need this if it's not pointing to anything?

dalarose commented on Assets/Scripts/SoundManager.cs:
```diff
@@ -5,11 +5,11 @@ public class SoundManager : MonoBehaviour {
-    //public AudioClip crincleAudioClip;
-    public AudioSource crincle;
+    public AudioClip crincleAudioClip;
+    AudioSource crincle;
```
let's mark properties explicitly `private` if they are meant to be so.

dalarose commented on Assets/Scripts/SoundManager.cs:
```diff
@@ -5,11 +5,11 @@ public class SoundManager : MonoBehaviour {
 
 
-    /*void Awake()
+    void Awake()
```
same goes for functions -- let's mark them explicitly `private` if they are intended to be so.

dalarose commented on Assets/Scripts/SoundManager.cs:
```diff
@@ -5,11 +5,11 @@ public class SoundManager : MonoBehaviour {
 
-    //public AudioClip crincleAudioClip;
-    public AudioSource crincle;
+    public AudioClip crincleAudioClip;
```
does this need to be public? is this property accessed by any class outside of SoundManager? if not, we should make it a `private` with the `SerializedField` attribute ([reference documentation](https://docs.unity3d.com/ScriptReference/SerializeField.html))

dalarose commented on Assets/Scripts/MainMenu.cs:
```diff
@@ -9,15 +9,4 @@ public void LoadMainGame()
     {
         SceneManager.LoadScene(1);
```
so we should make an enum for something like this so that we aren't using hardcoded numbers throughout the codebase. the issue with using hardcoded numbers like this is that to another programmer they look completely random (i have no idea what `1` means or points to here and it would require additional digging on my part to find out, which is not ideal and quickly makes other programmers mad at you lol).

so you could handle this multiple ways. you could use an [enum](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum) like i suggested above, where it would something like

```
enum GameScenes {
  Main,
  Settings,
  [... etc]
}
```
and you would reference it here like
`SceneManager.LoadScene(GameScene.Main);`

you could also use [constants](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/constants). so you could do something like this:
```
private const int MainScene = 1;
```

and then reference it by doing

`SceneManager.LoadScene(MainScene);`

Note: i tend to prefer going the enum route, i find it to be cleaner in this kind of context.

neoimpulse committed:
Auto stash before merge of "develop" and "origin/main"

neoimpulse committed:
Updated MainMenu.cs to use constants

neoimpulse committed:
Merge pull request #4 from neoimpulse/testbranch

Revert "Revert "new pull request test""

neoimpulse committed:
Design/Implement Game Level

- Design and Implemented game levels with with and lose conditions.
- Added GameManager.cs to handle win-lose conditions. Updated UIManager.cs and ShapesManager.cs.

neoimpulse committed:
Updates to Pull Request 5

- Resolved pull request comments on GameManager.cs and UIManager.cs

neoimpulse committed:
Merge pull request #5 from neoimpulse/testbranch

Implement Game Levels with Win and Lose Conditions

neoimpulse committed:
Move the “Restart” and “Premade level”  buttons

Moved the “Restart” and “Premade level” buttons to an Options Menu overlay that the user can open and close from the game board.

neoimpulse committed:
Revert "Move the “Restart” and “Premade level”  buttons"

This reverts commit 5caf20174a6c7220a626f9a9a23af392d6d61943.

neoimpulse committed:
Merge branch 'testbranch2' into testbranch

neoimpulse committed:
Set up save system.

Design and implement save system.

neoimpulse committed:
Updates to Pull request 6

- Resolved pull request comments on UIManager.cs

neoimpulse committed:
Updates to Pull request 6

- Resolved pull request comments on UIManager.cs

neoimpulse committed:
update to pull request 6

update to pull request 6. removed update method from uimanager.cs

neoimpulse committed:
update to pull request 6

update to pull request 6. removed update method from uimanager.cs

neoimpulse committed:
Merge pull request #6 from neoimpulse/testbranch

Moved the "Restart" and "Premade level" buttons

neoimpulse committed:
Design/Implement ability to save level progress

Design and implement the ability to save level progress between game sessions. Current Score, Level, and current time are saved.

neoimpulse committed:
Merge remote-tracking branch 'origin/testbranch' into testbranch

neoimpulse committed:
Updates to Pull Request 7

updates to pull request 7

neoimpulse committed:
Updates to Pull Request 7

updated GameManager.cs

neoimpulse committed:
Merge pull request #7 from neoimpulse/testbranch

Design/Implement the ability to Save Level Progress
